misc/python/materialize/cli/ci_coverage_pr_report.py

```diff
@@ -95,41 +99,85 @@ def mark_covered_lines(lcov_file: str, coverage: Coverage) -> None:
             line_nr = int(line_str)
             hit = int(hit_str) if hit_str.isnumeric() else int(float(hit_str))
             if line_nr in coverage[file]:
-                coverage[file][line_nr] = (coverage[file][line_nr] or 0) + hit
+                if unittests:
+                    if not coverage[file][line_nr]:
+                        coverage[file][line_nr] = (coverage[file][line_nr] or 0) - hit
+                else:
+                    coverage[file][line_nr] = (coverage[file][line_nr] or 0) + hit
```

Comment:
Can you explain what happens here (maybe in a comment in the code)?
Is it that uncovered will be -1, only unit-test coverage 0, other coverage +1?

Reply:
I did explain it, but it's at the beginning of the file. 0 = uncovered, other coverage positive, unit test negative. Maybe would be cleaner to introduce a type for that.

Reply:
Ok!
Is my understanding correct that we need to include unit test coverage as the last step because otherwise hits may cancel hits from other coverage files? Or do I miss something here?

Reply:
Correct.

Reply:
Sounds slightly dangerous 😎 
We should at least document that (ignore if this is already the case).

Reply:
Added an assert